lms/validation/_helpers/_base.py

```diff
@@ -25,3 +31,20 @@ def __init__(self, request):
         # Storing context needed for serialization or deserialization in
         # self.context is a marshmallow convention.
         self.context = {"request": request}
+
+    def parse(self, *args, **kwargs):
+        """
+        Parse and validate the request.
+
+        Use this schema to parse and validate ``self.context["request"]`` and
+        either return the successfully parsed params or raise a validation
+        error.
+
+        :raise lms.validation.ValidationError: if the request isn't valid
+        """
+        return _PYRAMID_PARSER.parse(self, self.context["request"], *args, **kwargs)
+
+
+@_PYRAMID_PARSER.error_handler
+def _handle_error(error, _req, _schema, _status_code, _headers):
+    raise ValidationError(messages=error.messages) from error
```

Comment:
I'd like to try moving `_handle_error` and `_PYRAMID_PARSER` into `PyramidRequestSchema` itself and see if that works

Reply:
Done